easybuild/tools/toolchain/toolchain.py

```diff
@@ -1049,39 +1060,92 @@ def handle_sysroot(self):
+        """
+        Append prepocessor paths for given dependency root directory
+        """
+        header_dirs = ["include"]
+
+        if extra_dirs is None:
+            extra_dirs = ()
+        elif extra_dirs and isinstance(extra_dirs, (str,)):
+            extra_dirs = [extra_dirs]
+
+        try:
+            header_dirs.extend(extra_dirs)
+        except TypeError as err:
+            err_msg = f"_add_dependency_cpp_headers: given extra_dirs is not iterable: {extra_dirs}"
+            raise EasyBuildError(err_msg) from err
+        else:
+            header_dirs = nub(header_dirs)  # remove duplicates
+
+        # mode of operation is defined by search-path-cpp-headers option
+        # toolchain option has precedence over build option
+        cpp_headers_mode = DEFAULT_SEARCH_PATH_CPP_HEADERS
+        build_opt = build_option("search_path_cpp_headers")
+        if self.options.get("search-path-cpp-headers") is not None:
+            cpp_headers_mode = self.options.option("search-path-cpp-headers")
+            self.log.debug("search-path-cpp-headers set by toolchain option: %s", cpp_headers_mode)
+        elif build_opt is not None:
+            cpp_headers_mode = build_opt
+            self.log.debug("search-path-cpp-headers set by build option: %s", cpp_headers_mode)
+
+        if cpp_headers_mode not in SEARCH_PATH_CPP_HEADERS:
+            raise EasyBuildError(
+                "Unknown value selected for option search-path-cpp-headers: %s. Choose one of: %s",
+                cpp_headers_mode, ", ".join(SEARCH_PATH_CPP_HEADERS)
+            )
+
+        for env_var in SEARCH_PATH_CPP_HEADERS[cpp_headers_mode]:
+            self.log.debug("Adding header paths to toolchain variable '%s': %s", env_var, dep_root)
+            self.variables.append_subdirs(env_var, dep_root, subdirs=header_dirs)
+
+    def _add_dependency_linker_paths(self, dep_root, extra_dirs=None):
+        """
+        Append linker paths for given dependency root directory
+        """
+        lib_dirs = ["lib64", "lib"]
+
+        if extra_dirs is None:
+            extra_dirs = ()
+        elif extra_dirs and isinstance(extra_dirs, (str,)):
+            extra_dirs = [extra_dirs]
+
+        try:
+            lib_dirs.extend(extra_dirs)
+        except TypeError as err:
+            err_msg = f"_add_dependency_linker_paths: given extra_dirs is not iterable: {extra_dirs}"
+            raise EasyBuildError(err_msg) from err
+        else:
+            lib_dirs = nub(lib_dirs)  # remove duplicates
+
+        env_var = "LDFLAGS"
+        self.log.debug("Adding lib paths to toolchain variable '%s': %s", env_var, dep_root)
+        self.variables.append_subdirs(env_var, dep_root, subdirs=lib_dirs)
 
-        for root in roots:
-            self.variables.append_subdirs("CPPFLAGS", root, subdirs=cpp_paths)
-            self.variables.append_subdirs("LDFLAGS", root, subdirs=ld_paths)
+    def _unique_ordered_list_append(self, base, extra_dirs):
```

Comment:
Did you intend to use this but forgot? It seems to make sense to me to have the logic in a function instead of 2 places.

Reply:
yeah, I did and backpedaled.  Such a method should not go into `Toolchain`, but to something more generic like `easybuild.tools.utilities`. I now had the time to deal with that and https://github.com/easybuilders/easybuild-framework/pull/4645/commits/b714e65e124949ab64b177c8e57c1d4f55c4cf7c adds that method in its proper place.